src/storage/src/hummock/compactor.rs

```diff
@@ -449,10 +449,30 @@ impl Compactor {
         }
     }
 
-    /// Handle a compaction task and report its status to hummock manager.
+    /// Handles a compaction task and reports its status to hummock manager.
     /// Always return `Ok` and let hummock manager handle errors.
     pub async fn compact(context: Arc<CompactorContext>, compact_task: CompactTask) -> bool {
         use risingwave_common::catalog::TableOption;
+
+        // Set a watermark SST id to prevent full GC from accidentally deleting SSTs for in-progress
+        // write op. The watermark is invalidated when this method exits.
+        let tracker_id = match context.sstable_id_manager.add_watermark_sst_id(None).await {
+            Ok(tracker_id) => tracker_id,
+            Err(err) => {
+                tracing::warn!("Failed to track pending SST id. {:#?}", err);
+                return false;
+            }
+        };
+        let sstable_id_manager_clone = context.sstable_id_manager.clone();
+        scopeguard::guard(
```

Comment:
Dumb question: if we don't assign a variable to this statement, is it possible for the rust compiler to drop the scopeguard before this function ends?

Reply:
You're right. Thx